packages/amplify-cli/src/project-config-version-check.ts

```diff
@@ -31,116 +36,124 @@ export async function checkProjectConfigVersion(context: Context): Promise<void>
-            if (checkFileContent(fileString)) {
-              filesToUpdate.push(filePath);
-            }
-          }
-        });
-      });
-    });
+    const templateFileNames = glob.sync('**/*template.{yaml,yml,json}', globOptions);
+
+    for (const templateFileName of templateFileNames) {
+      const absolutePath = path.join(backendDirPath, templateFileName);
+      const fileContent = fs.readFileSync(absolutePath, 'utf8');
+
+      if (checkFileContent(fileContent)) {
+        filesToUpdate.push(templateFileName);
+      }
+    }
   }
 
   if (filesToUpdate.length > 0) {
-    let confirmed = context.input.options && context.input.options.yes;
-    confirmed = confirmed || (await promptForConfirmation(context, filesToUpdate));
+    const confirmed = context.input.options?.yes || (await promptForConfirmation(context, filesToUpdate));
 
     if (confirmed) {
-      filesToUpdate.forEach(filePath => {
-        let fileString = fs.readFileSync(filePath, 'utf8');
-        fileString = updateFileContent(fileString);
-        fs.writeFileSync(filePath, fileString, 'utf8');
-      });
+      for (const fileName of filesToUpdate) {
+        const absolutePath = path.join(backendDirPath, fileName);
+        const fileContent = fs.readFileSync(absolutePath, 'utf8');
+
+        const newFileContent = updateFileContent(fileContent);
+
+        fs.writeFileSync(absolutePath, newFileContent, 'utf8');
+      }
+
       context.print.info('');
-      context.print.success('NodeJS runtime version updated successfully to 10.x in all the CloudFormation templates.');
-      context.print.warning('Make sure the template changes are pushed to the cloud by "amplify push"');
+      context.print.success(
+        `Node.js runtime version successfully updated  to ${lambdaRuntimeVersion} in all the CloudFormation templates.`,
+      );
+      context.print.warning('Run “amplify push” to deploy the updated templates to the cloud.');
+
+      result = true;
     }
+  } else {
+    // No update means 'updated' for caller
+    result = true;
   }
-}
 
-function checkFileContent(fileString: string): boolean {
-  let result = false;
+  return result;
+}
 
-  for (let i = 0; i < prevLambdaRuntimeVersions.length; i++) {
-    if (fileString.includes(prevLambdaRuntimeVersions[i])) {
-      result = true;
-      break;
+function checkFileContent(fileContent: string): boolean {
+  for (const prevLambdaRuntimeVersion of prevLambdaRuntimeVersions) {
+    if (fileContent.includes(prevLambdaRuntimeVersion)) {
+      return true;
     }
   }
 
-  return result;
+  return false;
 }
 
 function updateFileContent(fileString: string): string {
   let result = fileString;
-  prevLambdaRuntimeVersions.forEach(prevVersion => {
-    result = result.replace(prevVersion, lambdaRuntimeVersion);
-  });
+
+  for (const prevLambdaRuntimeVersion of prevLambdaRuntimeVersions) {
+    result = result.replace(prevLambdaRuntimeVersion, lambdaRuntimeVersion);
```

Comment:
Isn't this dangerous to replace all the strings matching? Cant we parse the file and update only the `Runtime` instead of blind string replace

Reply:
Enhanced it to be stricter and only update the `Runtime` property.